awx/main/models/ha.py

```diff
@@ -174,13 +177,19 @@ def refresh_capacity_fields(self):
         self.mem_capacity = get_mem_effective_capacity(self.memory)
         self.set_capacity_value()
 
-    def save_health_data(self, version, cpu, memory, last_seen=None, has_error=False):
+    def save_health_data(self, version, cpu, memory, uuid=None, last_seen=None, has_error=False):
         update_fields = []
 
         if last_seen is not None and self.last_seen != last_seen:
             self.last_seen = last_seen
             update_fields.append('last_seen')
 
+        if uuid is not None and self.uuid != uuid:
+            if self.uuid is not None:
+                logger.warn(f'Self-reported uuid of {self.hostname} changed from {self.uuid} to {uuid}')
```

Comment:
Know this isn't the heart of the PR, but love that you thought to surface this info in the logger 👍 . Can see this kind of thing (uuid-change) being responsible for some kind of esoteric bug down the line.